kairon/shared/data/processor.py

```
@@ -8717,10 +8717,26 @@ def save_doc_content(self, bot: Text, user: Text, doc_content, table_name: Text,
         for row in reversed(doc_content):
+            for column, value in row.items():
+                if column in metadata_map:
+                    data_type = metadata_map[column]
+                    try:
+                        if data_type == 'int':
+                            row[column] = int(value) if value else None
+                        elif data_type == 'float':
+                            row[column] = float(value) if value else None
+                    except (ValueError, TypeError):
+                        raise ValueError(
+                            f"Error converting column '{column}' with value '{value}' to type '{data_type}'")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)